cpp/src/datetime/datetime_ops.cu

```diff
@@ -648,6 +628,26 @@ std::unique_ptr<column> extract_second(column_view const& column, rmm::device_as
   return detail::extract_second(column, cudf::get_default_stream(), mr);
 }
 
+std::unique_ptr<cudf::column> extract_datetime_component(cudf::column_view const& column,
+                                                         datetime_component component,
+                                                         rmm::mr::device_memory_resource* mr)
```

Comment:
Add a stream to the public API.

Reply:
And accept a `device_async_resource_ref`

```suggestion
                                                         rmm::device_async_resource_ref mr)
```